src/uu/split/src/platform/unix.rs

```diff
@@ -107,19 +109,46 @@ impl Drop for FilterWriter {
 pub fn instantiate_current_writer(
     filter: &Option<String>,
     filename: &str,
-) -> BufWriter<Box<dyn Write>> {
+) -> Result<BufWriter<Box<dyn Write>>> {
     match filter {
-        None => BufWriter::new(Box::new(
+        None => Ok(BufWriter::new(Box::new(
             // write to the next file
             std::fs::OpenOptions::new()
                 .write(true)
                 .create(true)
+                .truncate(true)
                 .open(std::path::Path::new(&filename))
-                .unwrap(),
-        ) as Box<dyn Write>),
-        Some(ref filter_command) => BufWriter::new(Box::new(
+                .map_err(|_| {
+                    Error::new(
+                        ErrorKind::Other,
+                        format!("unable to open '{}'; aborting", filename),
+                    )
+                })?,
+        ) as Box<dyn Write>)),
+        Some(ref filter_command) => Ok(BufWriter::new(Box::new(
             // spawn a shell command and write to it
-            FilterWriter::new(filter_command, filename),
-        ) as Box<dyn Write>),
+            FilterWriter::new(filter_command, filename)?,
+        ) as Box<dyn Write>)),
+    }
+}
+
+pub fn paths_refer_to_same_file(p1: &str, p2: &str) -> bool {
```

Comment:
[paths_refer_to_same_file ](https://uutils.github.io/coreutils-docs/dev/uucore/fs/fn.paths_refer_to_same_file.html) seems to be the same, no ?

Reply:
Unfortunately, that doesn't handle the "-" case where we are trying to tell if something like
`split < xaa` is run. This is the reason it needs to compare to /dev/fd/0. There doesn't seem to be any other way to handle this case correctly (which is part of the gnu test suite).
```
$ echo 12345 > xaa && split < xaa
split: 'xaa' would overwrite input; aborting
```
It might be worth extending that method to handle this case, but I'm not sure if this is something we generally want to do. With the code as I have it here it handles that case correctly.

Reply:
or you keep  the "-" mgmt and plug the rest of the function on paths_referr_to_same_file ?

Reply:
thanks